### PR TITLE
Adding keyword extraction API

### PR DESCRIPTION
**Pull request by rreece**
Adds keyword extraction API.

Issue:
https://github.com/tenstorrent/edenai/issues/76

### CODE REVIEW

File: edenai_apis/apis/tenstorrent/outputs/text/sentiment_analysis_output.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-    "standardized_response": {
-        "general_sentiment": "positive",
-        "general_sentiment_rate": 0.984375,
-        "items": []
-    }
-}
+  "original_response": {
+    "confidence": 0.37890625,
+    "predicted_class": 0,
+    "prediction": "negative"
```
Comment (tstescoTT) [inline edenai_apis/apis/tenstorrent/outputs/text/sentiment_analysis_output.json:5]: Do we know why the sentiment analysis flipped? Did the test text change?

File: edenai_apis/apis/tenstorrent/tenstorrent_api.py [hunk 2]
```diff
@@ -17,10 +17,11 @@ def __init__(self, api_keys: Dict = {}):
             ProviderDataEnum.KEY, provider_name=self.provider_name, api_keys = api_keys
         )
         self.api_key = self.api_settings["api_key"]
-        self.base_url = "https://sentiment-analysis--eden-ai.workload.tenstorrent.com"
+#        self.base_url = self.api_settings["base_url"]
+#        self.base_url = "https://sentiment-analysis--eden-ai.workload.tenstorrent.com"
```
Comment (tstescoTT) [inline edenai_apis/apis/tenstorrent/tenstorrent_api.py]: Can you removed the commented out URLs?

File: edenai_apis/apis/tenstorrent/tenstorrent_text_api.py [hunk 3]
```diff
@@ -9,10 +12,37 @@
 
 
 class TenstorrentTextApi(TextInterface):
+
+    def text__keyword_extraction(
+        self, language: str, text: str
+    ) -> ResponseType[KeywordExtractionDataClass]:
+        base_url = "https://keyword-extraction--eden-ai.workload.tenstorrent.com"
```
Comment (tstescoTT) [inline edenai_apis/apis/tenstorrent/tenstorrent_text_api.py:19]: LGTM

File: edenai_apis/apis/tenstorrent/tenstorrent_text_api.py [hunk 4]
```diff
@@ -9,10 +12,37 @@
...
+        return ResponseType[KeywordExtractionDataClass](
+            original_response=original_response,
+            standardized_response=standardized_response,
+        )
+
     def text__sentiment_analysis(
         self, language: str, text: str
     ) -> ResponseType[SentimentAnalysisDataClass]:
-        url = f"{self.base_url}/predictions/sentiment_analysis"
+        base_url = "https://sentiment-analysis--eden-ai.workload.tenstorrent.com"
```
Comment (tstescoTT) [inline edenai_apis/apis/tenstorrent/tenstorrent_text_api.py:44]: LGTM

**Comment by tstescoTT** [comment]
Are we removing the `base_url` from https://github.com/tenstorrent/edenai-apis/blob/ed639b4166d2427dc58c0faa1ba3ccce1dda137d/edenai_apis/api_keys/tenstorrent_settings_template.json#L2?

**Comment by tstescoTT** [comment]
Running both sentiment analysis and keyword extraction pytests pass for me:
```
pytest -s edenai_apis/tests/outputs.py --provider tenstorrent --feature text --subfeature keyword_extraction
pytest -s edenai_apis/tests/outputs.py --provider tenstorrent --feature text --subfeature sentiment_analysis
```

**Comment by rreece** [comment]
removed the base_url from the template

**Review by tstescoTT — COMMENTED** [review]
Approved!

**Pull request merged**